src/content/docs/serverless-function-monitoring/aws-lambda-monitoring/instrument-lambda-function/instrument-your-own.mdx

```diff
@@ -0,0 +1,262 @@
+          Layers:
+            # This layer includes the New Relic Lambda extension, a sidecar process that sends telemetry,
+            # as well as the New Relic agent for Node.js, and a handler wrapper that makes integration easy.
+            - !Sub arn:${AWS::Partition}:lambda:${AWS::Region}:451483290750:layer:NewRelicNodeJS12X:34
+          Policies:
+            # This policy allows the lambda to know the value of the New Relic license key. We need this so
+            # that we can send telemetry back to New Relic
+            - AWSSecretsManagerGetSecretValuePolicy:
+                SecretArn: !ImportValue NewRelicLicenseKeySecret-NewRelic-LicenseKeySecretARN
+    ```
+
+    Conventionally, you'll have a file named `template.yaml` that describes your function, and its resources.
+  </Collapser>
+
+  <Collapser
+    id="serverless"
+    title="Serverless Framework"
+  >
+Serverless Framework is a popular development and deployment tool for serverless applications. It's written for AWS in Node.js, and acts mostly as a high level abstraction on top of CloudFormation templates. It works well for Node, Python, Ruby, Java, and .NET functions.
+
+New Relic offers a [Serverless Framework Plugin](https://github.com/newrelic/serverless-newrelic-lambda-layers) to simplify instrumentation of your Serverless Framework application.
+
+To install the plugin, run this command:
+
+```bash
+    npm install --save-dev serverless-newrelic-lambda-layers
+```
+
+Or, alternatively, you can run this command:
+
+```bash
+    yarn add --dev serverless-newrelic-lambda-layers
+```
+
+Then, find your [New Relic Account ID](/docs/accounts/install-new-relic/account-setup/account-id), your [New Relic Personal API Key](/docs/apis/get-started/intro-apis/types-new-relic-api-keys#personal-api-key)
+
+Now add the following to your `serverless.yaml` file:
+
+```yaml
+    plugins:
+      - serverless-newrelic-lambda-layers
+    custom:
+      newRelic:
+        accountId: your-new-relic-account-id-here
+        apiKey: your-new-relic-personal-api-key-here
+        linkedAccount: your-new-relic-integration-account-name-here
+    ```
+  </Collapser>
+
+  <Collapser
+    id="terraform"
+    title="Terraform"
+  >
+    Terraform is a popular general-purpose infrastructure as code tool. It can be used to manage AWS resources. We offer [some examples](https://github.com/newrelic/newrelic-lambda-extension/tree/main/examples/terraform) of New Relic instrumented Lambda functions deployed using Terraform scripts.
+  </Collapser>
+
+  <Collapser
+    id="console"
+    title="AWS manual instrumentation"
+  >
+
+
+While it is more error prone and labor intensive than the approaches above, it's possible to manually alter the configuration of a Lambda function to use New Relic from the AWS Lambda Console, for Node.js, Python, Ruby and Java.
+
+Here's an example for how to instrument New Relic Lambda monitoring for a Ruby runtime:
```

Comment:
@fallwith would be a good person to review the Ruby-specific instructions.

Reply:
Reviewed with @fallwith !